web/src/api.ts

```diff
@@ -48,6 +49,29 @@ export const getListing = async (listingId: number): Promise<Listing> => {
   return res.json();
 };
 
+/**
+ * Retrieves merchant with the given email from the database.
+ */
+export const getMerchantWithEmail = async (
+  email: string
+): Promise<MerchantResponse> => {
+  const res = await fetch(
+    `${process.env.REACT_APP_SERVER_URL}/merchants?email=${email}`
```

Comment:
did we add index on email in datastore?
why not get merchant by firebase uid?

Reply:
@ifhuang 
For the first question: Currently, all the properties are indexed by default in datastore. I'm planning to create an index.yaml file to specify which ones to be indexed.

For the second question: I'm planning to reuse this function for sign up, to check for the case where an account with this email is created in Firebase but not in the database. This means that the sign up process wasn't successfully completed and we should delete the account from Firebase and retry signing up for that user.